Source/SwiftLintBuiltInRules/Rules/Lint/AsyncWithoutAwaitRuleExamples.swift

```diff
@@ -0,0 +1,338 @@
+            for await foo in bar {}
+        }
+        """),
+        Example("""
+        func test() async {
+            while let foo = await bar() {}
+        }
+        """),
+        Example("""
+        func test() async {
+            async let foo = bar()
+            let baz = await foo
+        }
+        """),
+        Example("""
+        func test() async {
+            async let foo = bar()
+            await foo
+        }
+        """),
+        Example("""
+        func test() async {
+            async let foo = bar()
+        }
+        """),
+        Example("func foo(bar: () async -> Void) { { } }"),
+        Example("func foo(bar: () async -> Void = { await baz() }) { {} }"),
+        Example("func foo() -> (() async -> Void)? { {} }"),
+        Example("""
+        func foo(
+            bar: () async -> Void,
+            baz: () -> Void = {}
+        ) { { } }
+        """),
+    ]
+
+    static let triggeringExamples = [
+        Example("""
+        func test() ↓async {
+            perform()
+        }
+        """),
+        Example("""
+        func test() {
+            func baz() ↓async {
+                qux()
+            }
+            perform()
+            func baz() {
+                qux()
+            }
+        }
+        """),
+        Example("""
+        func test() ↓async {
+            func baz() async {
+                await qux()
+            }
+        }
+        """),
+        Example("""
+        func test() ↓async {
+          func foo() ↓async {}
+          let bar = { await foo() }
+        }
+        """),
+        Example("""
+        func test() ↓async {
+            let bar = {
+                func foo() ↓async {}
+            }
+        }
+        """),
+        Example("let x: (() ↓async -> Void)? = { test() }"),
+        Example("""
+        var test: Int {
+            get ↓async throws {
+                foo()
+            }
+        }
+        """),
+        Example("""
+        var test: Int {
+            get ↓async throws {
+                func foo() ↓async {}
+                let bar = { await foo() }
+            }
+        }
+        """),
+        Example("""
+        var test: Int {
+            get throws {
+                func foo() {}
+                let bar: () ↓async -> Void = { foo() }
+            }
+        }
+        """),
+        Example("init() ↓async {}"),
+        Example("""
+        init() ↓async {
+            func foo() ↓async {}
+            let bar: () ↓async -> Void = { foo() }
+        }
+        """),
+        Example("""
+        subscript(row: Int) -> Double {
+            get ↓async {
+                1.0
+            }
+        }
+        """),
+        Example("""
+        func test() ↓async {
+            for foo in bar {}
+        }
+        """),
+        Example("""
+        func test() ↓async {
+            while let foo = bar() {}
+        }
+        """),
+        Example("func foo(bar: () ↓async -> Void = {}) { }"),
```

Comment:
This looks strange. For a parameter, only because the default value isn't `async` isn't a sufficient indicator that the whole closure doesn't need to be `async`.

Reply:
So what do you suggest? Ignore it if it is a default argument.